tests/dragonfly/replication_test.py

```diff
@@ -23,30 +23,57 @@
 # 1. Number of master threads
 # 2. Number of threads for each replica
 # 3. Seeder config
+# 4. Admin port
 replication_cases = [
-    (8, [8], dict(keys=10_000, dbcount=4)),
-    (6, [6, 6, 6], dict(keys=4_000, dbcount=4)),
-    (8, [2, 2, 2, 2], dict(keys=4_000, dbcount=4)),
-    (4, [8, 8], dict(keys=4_000, dbcount=4)),
-    (4, [1] * 8, dict(keys=500, dbcount=2)),
-    (1, [1], dict(keys=100, dbcount=2)),
+    (8, [8], dict(keys=10_000, dbcount=4), False),
+    (6, [6, 6, 6], dict(keys=4_000, dbcount=4), False),
+    (8, [2, 2, 2, 2], dict(keys=4_000, dbcount=4), False),
+    (4, [8, 8], dict(keys=4_000, dbcount=4), False),
+    (4, [1] * 8, dict(keys=500, dbcount=2), False),
+    (1, [1], dict(keys=100, dbcount=2), False),
+    (8, [8], dict(keys=10_000, dbcount=4), True),
+    (6, [6, 6, 6], dict(keys=4_000, dbcount=4), True),
+    (8, [2, 2, 2, 2], dict(keys=4_000, dbcount=4), True),
+    (4, [8, 8], dict(keys=4_000, dbcount=4), True),
+    (4, [1] * 8, dict(keys=500, dbcount=2), True),
+    (1, [1], dict(keys=100, dbcount=2), True),
 ]
 
 
 @pytest.mark.asyncio
-@pytest.mark.parametrize("t_master, t_replicas, seeder_config", replication_cases)
+@pytest.mark.parametrize("t_master, t_replicas, seeder_config, from_admin_port", replication_cases)
 async def test_replication_all(
-    df_local_factory, df_seeder_factory, t_master, t_replicas, seeder_config
+    df_local_factory, df_seeder_factory, t_master, t_replicas, seeder_config, from_admin_port
 ):
-    master = df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
-    replicas = [
-        df_local_factory.create(port=BASE_PORT + i + 1, proactor_threads=t)
-        for i, t in enumerate(t_replicas)
-    ]
+    master = (
+        df_local_factory.create(port=BASE_PORT, proactor_threads=t_master)
+        if not from_admin_port
+        else df_local_factory.create(
+            port=BASE_PORT, admin_port=ADMIN_PORT, proactor_threads=t_master
+        )
+    )
+    replicas = (
+        [
+            df_local_factory.create(port=BASE_PORT + i + 1, proactor_threads=t)
+            for i, t in enumerate(t_replicas)
+        ]
+        if not from_admin_port
```

Comment:
I am not particular fun of this but even if we extract it as a function, we still have to pass a bunch of arguments making it quite verbose. 

@adiholden is there a more pythonic way to do this and avoid the verbosity? 

1. If we extract the changes to a function, it's still amost the same + you have to pass a bunch of arguments on the caller side. 
2. If we copy paste the body of the test into a separate function, it's a redundant copying of 90% of the same test case.